enterprise/cluster/src/main/java/org/neo4j/cluster/client/ClusterJoin.java

```diff
@@ -134,83 +128,53 @@ private void joinByConfig() throws TimeoutException
         }
         else
         {
-            URI[] memberURIs = Iterables.toArray(URI.class,
-                    Iterables.map( new Function<HostnamePort, URI>()
+            URI[] memberURIs = Iterables.toArray( URI.class,
+                    Iterables.map( new Function<HostnamePort,URI>()
                     {
                         @Override
                         public URI apply( HostnamePort member )
                         {
                             return URI.create( "cluster://" + resolvePortOnlyHost( member ) );
                         }
-                    }, hosts));
+                    }, hosts ) );
 
-            while( true )
+            while ( true )
             {
                 userLog.info( "Attempting to join cluster of %s", hosts.toString() );
                 Future<ClusterConfiguration> clusterConfig =
                         cluster.join( this.config.getClusterName(), memberURIs );
 
-                if (config.getClusterJoinTimeout() > 0)
+                try
                 {
-                    try
-                    {
-                        userLog.info( "Joined cluster: %s", clusterConfig.get( config.getClusterJoinTimeout(), TimeUnit.MILLISECONDS ) );
-                        return;
-                    }
-                    catch ( InterruptedException e )
-                    {
-                        userLog.warn( "Could not join cluster, interrupted. Retrying..." );
-                    }
-                    catch ( ExecutionException e )
-                    {
-                        messagesLog.debug( "Could not join cluster " + this.config.getClusterName() );
-                        if ( e.getCause() instanceof IllegalStateException )
-                        {
-                            throw ((IllegalStateException) e.getCause());
-                        }
-
-                        if ( config.isAllowedToCreateCluster() )
-                        {
-                            // Failed to join cluster, create new one
-                            userLog.info( "Could not join cluster of %s", hosts.toString() );
-                            userLog.info( "Creating new cluster with name [%s]...", config.getClusterName() );
-                            cluster.create( config.getClusterName() );
-                            break;
-                        }
-
-                        userLog.warn( "Could not join cluster, timed out. Retrying..." );
-                    }
-                } else
+                    ClusterConfiguration clusterConf =
+                            config.getClusterJoinTimeout() > 0
+                            ? clusterConfig.get( config.getClusterJoinTimeout(), TimeUnit.MILLISECONDS )
+                            : clusterConfig.get();
+                    userLog.info( "Joined cluster: %s", clusterConf );
+                    return;
+                }
+                catch ( InterruptedException e )
```

Comment:
This was not introduced now but it looks suspicious that we'll spin in the `while(true)` loop and would not react to interrupts. I think this might cause database to not shutdown properly.